### ISSUE TITLE
prefer-dot-shorthands doesn't support switch-case and unnamed constructors

### ISSUE THREAD

**Issue by ngoc-quoc-huynh** [bancolombia/dart-code-linter#265]
The new introduced rule `prefer-dot-shorthands` currently doesn't support two cases. According to the [docs](https://dart.dev/language/dot-shorthands), they should be supported.

- [Enums in switch-case](https://dart.dev/language/dot-shorthands#enums)
```dart
// Doesn't trigger warning
String colorCode(LogLevel level) {
  // Use dot shorthand syntax for enum values in switch cases:
  return switch (level) {
    LogLevel.debug => 'gray',
    LogLevel.info => 'blue',
    LogLevel.warning => 'orange', 
    LogLevel.error => 'red',
  };
```

- [Unnamed constructors](https://dart.dev/language/dot-shorthands#unnamed-constructors)
```dart
// Doesn't trigger warning
class _PageState extends State<Page> with SingleTickerProviderStateMixin {
  late final AnimationController _animationController = AnimationController(
    vsync: this,
  );
  final ScrollController _scrollController = ScrollController();

  final GlobalKey<ScaffoldMessengerState> scaffoldKey =
      GlobalKey<ScaffoldMessengerState>();

  Map<String, Map<String, bool>> properties = <String, Map<String, bool>>{};
// ...
}

// Expected code
class _PageState extends State<Page> {
  late final AnimationController _animationController = .new(vsync: this);
  final ScrollController _scrollController = .new();
  final GlobalKey<ScaffoldMessengerState> scaffoldKey = .new();
  Map<String, Map<String, bool>> properties = .new();
  // ...
}
```

**Comment by ngoc-quoc-huynh** [comment]
@juan-campuzano I checked the code and `unnamed constructors` is intentionally not included, any reason why?
